general/device.cpp

```diff
@@ -181,6 +178,8 @@ Device::~Device()
    Get().host_mem_class = MemoryClass::HOST;
    Get().device_mem_type = MemoryType::HOST;
    Get().device_mem_class = MemoryClass::HOST;
+   Get().device_temp_mem_type = MemoryType::HOST;
+   Get().device_temp_mem_class = MemoryClass::HOST;
```

Comment:
I think it would make sense to get rid of the `device_temp_mem_type` and everything related to it and just keep the `device_temp_mem_class`. The assumption with these changes is that the `DeviceMemorySpace` can optionally support temporary memory but that the temporary `MemoryType` is _not_ distinct from the permanent `MemoryType`. What do others think?

We could allow for combinations for permanent and temporary device memory types but that brings back the need for a `TEMP` `MemoryType` for each device `MemoryType` that has a temp option.